language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -617,9 +618,8 @@ impl LibraVM {
         }
 
         // Record the histogram count for transactions per block.
-        match i64::try_from(count) {
-            Ok(val) => BLOCK_TRANSACTION_COUNT.set(val),
-            Err(_) => BLOCK_TRANSACTION_COUNT.set(std::i64::MAX),
+        if let Some(val) = f64::from_usize(count) {
+            BLOCK_TRANSACTION_COUNT.observe(val)
```

Comment:
I would just avoid the extra dependency and just do an `as` conversion:

`BLOCK_TRANSACTION_COUNT.observe(count as f64)`

This conversion wont ever fail, but will loose precision in the process (but IMO that's fine for something like this)

Reply:
Sounds fine to me